src/common/libflux/mrpc.c

```diff
@@ -526,29 +527,21 @@ flux_mrpc_t *flux_mrpc_pack (flux_t *h,
 
 void *flux_mrpc_aux_get (flux_mrpc_t *mrpc, const char *name)
 {
-    assert (mrpc->magic == MRPC_MAGIC);
-    if (!mrpc->aux)
+    if (!mrpc) {
+        errno = EINVAL;
         return NULL;
-    return zhash_lookup (mrpc->aux, name);
+    }
+    return aux_get (mrpc->aux, name);
 }
 
 int flux_mrpc_aux_set (flux_mrpc_t *mrpc, const char *name,
                       void *aux, flux_free_f destroy)
 {
-    assert (mrpc->magic == MRPC_MAGIC);
-    if (!mrpc->aux)
-        mrpc->aux = zhash_new ();
-    if (!mrpc->aux) {
-        errno = ENOMEM;
-        return -1;
-    }
-    zhash_delete (mrpc->aux, name);
-    if (zhash_insert (mrpc->aux, name, aux) < 0) {
-        errno = ENOMEM;
+    if (!mrpc) {
```

Comment:
don't know if it should be in this PR or perhaps we can just create a new issue for it.  But there are several other places in ```mrpc.c``` that do ```assert (mrpc->magic == MRPC_MAGIC)``` instead of ```if (!mrpc)```.  It should probably be consistent.

Reply:
I'd say that's a different issue.